analyzer/file/png/main.zeek

```diff
@@ -0,0 +1,106 @@
+module PNG;
+
+export {
+
+	redef enum Log::ID += { LOG };
+
+	const PNG_COLOUR_TYPE_GREYSCALE = 0;
+	const PNG_COLOUR_TYPE_TRUECOLOUR = 2;
+	const PNG_COLOUR_TYPE_INDEXED_COLOUR = 3;
+	const PNG_COLOUR_TYPE_GREYSCALE_WITH_ALPHA = 4;
+	const PNG_COLOUR_TYPE_TRUECOLOUR_WITH_ALPHA = 6;
+
+	## Mapping between numeric codes and string values for colour types
+	const colour_types: table[count] of string = {
+		[PNG_COLOUR_TYPE_GREYSCALE] = "greyscale",
+		[PNG_COLOUR_TYPE_TRUECOLOUR] = "truecolour",
+		[PNG_COLOUR_TYPE_INDEXED_COLOUR] = "indexed-colour",
+		[PNG_COLOUR_TYPE_GREYSCALE_WITH_ALPHA] = "greyscale with alpha",
+		[PNG_COLOUR_TYPE_TRUECOLOUR_WITH_ALPHA] = "truecolour with alpha"
+	} &default=function(i: count):string { return fmt("unknown-%d", i); };
+
+	type Info: record {
+		## Current timestamp
+		ts:             time             &log;
+		## File ID of this PNG
+		id:             string           &log;
+		## Chunk types in the PNG, in the order in which they appeared
+		chunks:         vector of string &log &optional;
+		## Image width in pixels
+		width:          count            &log &optional;
+		## height in pixels
+		height:         count            &log &optional;
+		## Image colour type
+		colour_type:    string           &log &optional;
+		## Image bit depth
+		bit_depth:      count            &log &optional;
+		## Flag is set to true if image is interlaced
+		interlaced:     bool             &log &optional;
+		## Last modification time
+		last_modified:  time             &log &optional;
+	};
+
+	global log_png: event(rec: Info);
+
+	global set_file: hook(f: fa_file);
+
+	type PaletteEntry: record {
+		red: count;
+		green: count;
+		blue: count;
+	};
+}
+
+redef record fa_file += {
+	png: Info &optional;
+};
+
+const png_mime_types = { "image/png" };
+
+event zeek_init() &priority=5
+	{
+	#Files::register_for_mime_types(Analyzer::get_tag("SPICY_PNG"), png_mime_types);
```

Comment:
So this (or some of this) is still not working?

Reply:
It's not necessary - the evt file has the mime type. So - I should just remove this, sorry.